src/Codeception/Lib/Driver/Db.php

```diff
@@ -290,11 +290,13 @@ public function executeQuery($query, array $params): PDOStatement
         $i = 0;
         foreach ($params as $param) {
             ++$i;
-            if (is_bool($param)) {
+            if (is_null($param)) {
+                $type = PDO::PARAM_NULL;
+            } elseif (is_bool($param)) {
                 $type = PDO::PARAM_BOOL;
             } elseif (is_int($param)) {
                 $type = PDO::PARAM_INT;
-            } elseif ($this->isBinary($param)) {
+            } elseif (is_string($param) && $this->isBinary($param)) {
```

Comment:
is_string($param) is really not needed here, as the method mb_detect_encoding() is always gonna cast the first parameter to (string). 

Maybe the method isBinary() should be completly replaced with its contents, because currently that method does not detect a binary string or anything binary related.

I suggest using `!mb_detect_encoding($string, null, true)` directly here.

Reply:
In our case, the arg is null because our test is trying to find rows where a column has a null value.  Please merget his.

Reply:
@troy-rudolph 

did you try to use the suggestion mentioned by gurau?

Reply:
@eerison I tried, solution suggested from @guraurobi is incorrect: 
`[TypeError] mb_detect_encoding(): Argument #1 ($string) must be of type string, null given`
`is_string` do the trick here

Reply:
Yep that is true `mb_detect_encoding` requires a string.

> as the method mb_detect_encoding() is always gonna cast the first parameter to (string).

@guraurobi it gonna cast if you are not using `declare(strict_types=1)`, otherwise it will throw the exception mentioned by @Daredzik .

> Maybe the method isBinary() should be completly replaced with its contents, because currently that method does not detect a binary string or anything binary related.

IMO it is not related with this issue, i guess we could implement this in other PR, I don't see any issue with the current code, and as we can see it was already tested by a lot of guys 😄.

Then I guess we could mark this thread as solved, don't you @guraurobi ?

Btw: @Daredzik could you confirm if others suggestions from @guraurobi worked or not?